src/shims/env.rs

```diff
@@ -50,8 +50,8 @@ impl<'tcx> EnvVars<'tcx> {
         if ecx.machine.communicate() || !config.forwarded_env_vars.is_empty() {
             for (name, value) in env::vars_os() {
                 let forward = match ecx.machine.communicate() {
-                    true => !excluded_env_vars.iter().any(|v| v.as_str() == &name),
-                    false => config.forwarded_env_vars.iter().any(|v| v.as_str() == &name),
+                    true => !excluded_env_vars.iter().any(|v| **v == name),
+                    false => config.forwarded_env_vars.iter().any(|v| **v == name),
```

Comment:
FWIW I find the `as_str` version here easier to write since I don't have to figure out how many `*` need to be changed.

But in terms of readability there's no significant difference, so I am fine with this diff. (But if we add clippy to CI, I'd ask for whatever lint triggered this to be `allow`ed.)

Reply:
I don't think it was a lint. It was just some drive by refactoring. The lint actually suggested to deref the result of the as_str